QtAwesome/iconic_font.py

```diff
@@ -1,22 +1,81 @@
 """Classes handling iconic fonts"""
 
 from __future__ import print_function
-from PyQt4.QtCore import Qt, QObject, QPoint, QRect, qRound, QByteArray
+from PyQt4.QtCore import Qt, QObject, QPoint, QRect, qRound, QByteArray, QTimer
 from PyQt4.QtGui import (QIcon, QColor, QIconEngine, QPainter, QPixmap,
                          QFontDatabase, QFont)
 import json
 import os
 
+
 _default_options = {
     'color': QColor(50, 50, 50),
     'scale_factor': 0.9,
 }
 
 
-class CharIconPainter:
+class CharIconAnimator:
```

Comment:
*  Change name to CharIconRotation :+1: 
* As I said before spin and pulse are the names used in font awesome, that is why I left the same names, so pulse and spin refer to font awesome `fa-spin fa-pulse`

Some icons allow for a pulsing motion, and any icon could be used for spinning (plus I can add direction: clockwise, counterclockwise) some have a sniping motion. By default we should handle these `types` without needing to define the step and the intervals manually. Maybe in Gitter you can explain me :)